libs/jump_portrait/src/jump_portrait/fetch.py

```diff
@@ -119,6 +120,52 @@ def get_jump_image(
     return result
 
 
+def get_jump_image_iter(metadata: pl.DataFrame, channel: List[str],
+                        site: List[str], correction:str='Orig',
+                        print_progress: bool=True,
+                        ) -> (pl.DataFrame, List[tuple]):
+    '''
+       Load jump image associated to metadata in a threaded fashion.
+        ----------
+    Parameters:
+        metadata(pl.DataFrame): must have the shape ("Metadata_Source",
+    "Metadata_Batch", "Metadata_Plate", "Metadata_Well")
+        channel(List[str]): list of channel desired
+            Must be in ['DNA', 'ER', 'AGP', 'Mito', 'RNA']
+        site(List[str]): list of site desired
+            For compound, must be in ['1' - '6']
+            For ORF, CRISPR, must be in ['1' - '9']
+        correction(str): Must be 'Illum' or 'Orig'
+        print_progress(bool): Whether to enable tqdm or not.
+        ----------
+    Return:
+        features(pl.DataFrame): DataFrame collecting the metadata, channel, site,
+    correction + the image
+        work_fail(List(tuple): List collecting tuple of metadata which
+    failed to load an image
+
+    '''
+    iterable = [(*metadata.row(i), ch, s, correction)
+               for i in range(metadata.shape[0]) for s in site for ch in channel]
+    img_list = parallel(iterable, batch_processing(try_function(get_jump_image)),
+                        print_progress=print_progress)
+     
+    img_list = sorted(img_list, key=lambda x: len(x))
+    fail_success = {k: list(g) for k, g in groupby(img_list, key=lambda x: len(x))}
+    if len(fail_success) == 1:
+        img_success = list(fail_success.values())[0]
+        work_fail = []
+    else:
+        work_fail, img_success = fail_success.values()
+    features = pl.DataFrame(img_success,
+                               schema=["Metadata_Source", "Metadata_Batch", "Metadata_Plate", "Metadata_Well",
+                                        "channel", "site", "correction",
+                                        "img"])
```

Comment:
Do not put numpy arrays inside DataFrames. If you want to return a set of data+metadata, return them as tuples. Normally I would suggest to do so by stacking all images but I know they don't all have the same size so let us use a tuple of image,meta pairs. A Dataframe is an overkill for this. Just specify what metadata is included in the output within a comment by the end of the function.

Reply:
My main motivation for using a Dataframe is that it allows me to do some grouping if I need to, or order things easily. But understood, I will remove that